packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -247,42 +247,99 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
     super.didChangeNext(nextRoute);
   }
 
+  // A callback method that disposes existing train hopping animation and
+  // removes its listener.
+  //
+  // This field is non-null if there is a train hopping in progress.
+  VoidCallback _trainHoppingListenerRemover;
+
   void _updateSecondaryAnimation(Route<dynamic> nextRoute) {
+    // There is an existing train hopping in progress. Unfortunately, we cannot
```

Comment:
How do you know at this point that there's an existing train hopping in progress?

Reply:
whoever calls this method will need to make sure to set this private field to null as to indicate there is no more train hopping in progress. I think the comment in line 253 is not clear enough, will update